api_app/api/routes/workspace_users.py

```diff
@@ -0,0 +1,68 @@
+from fastapi import APIRouter, Depends, Response, status
+from api.dependencies.workspaces import get_workspace_by_id_from_path
+from resources import strings
+from services.authentication import get_access_service
+from models.schemas.users import UsersInResponse, AssignableUsersInResponse
+from models.schemas.roles import RolesInResponse
+from services.authentication import get_current_admin_user, get_current_workspace_owner_or_researcher_user_or_airlock_manager_or_tre_admin
+
+workspaces_users_admin_router = APIRouter(dependencies=[Depends(get_current_admin_user)])
+workspaces_users_shared_router = APIRouter(dependencies=[Depends(get_current_workspace_owner_or_researcher_user_or_airlock_manager_or_tre_admin)])
+
+
+@workspaces_users_shared_router.get("/workspaces/{workspace_id}/users", response_model=UsersInResponse, name=strings.API_GET_WORKSPACE_USERS)
+async def get_workspace_users(workspace=Depends(get_workspace_by_id_from_path)) -> UsersInResponse:
+    access_service = get_access_service()
```

Comment:
Might be cleaner if get_access_service is a dependancy in the same way as the workspace.

```
@workspaces_users_shared_router.get("/workspaces/{workspace_id}/users", response_model=UsersInResponse, name=strings.API_GET_WORKSPACE_USERS)
async def get_workspace_users(workspace=Depends(get_workspace_by_id_from_path), access_service=Depends(get_access_service)) -> UsersInResponse:
    users = access_service.get_workspace_users(workspace)
    return UsersInResponse(users=users)
```

Reply:
I agree this is cleaner, the pattern was copied from an existing method, do you want me to refactor that on too.?

Reply:
Probably on of mine :) Yes, please do it its only that one instance.

Reply:
Have refactored the new bits but left the existing one (workspaces.py) in place.